diff --git a/object-translation-bundle/tests/Unit/ObjectTranslatorTest.php b/object-translation-bundle/tests/Unit/ObjectTranslatorTest.php
index cd9a0cc..833edad 100644
--- a/object-translation-bundle/tests/Unit/ObjectTranslatorTest.php
+++ b/object-translation-bundle/tests/Unit/ObjectTranslatorTest.php
@@ -17,6 +17,13 @@ class ObjectTranslatorTest extends TestCase
         $this->assertSame('value2', $object->prop2());
         $this->assertSame('value3', $object->getProp3());
     }
+
+    public function testCallUsesGetterIfAvailable()
+    {
+        $object = new TranslatedObject(new ObjectForTranslationStub());
+
+        $this->assertSame('value3', $object->prop3());
+    }
 }
 
 class ObjectForTranslationStub
