diff --git a/object-translation-bundle/tests/Unit/ObjectTranslatorTest.php b/object-translation-bundle/tests/Unit/ObjectTranslatorTest.php
index 833edad..65ba421 100644
--- a/object-translation-bundle/tests/Unit/ObjectTranslatorTest.php
+++ b/object-translation-bundle/tests/Unit/ObjectTranslatorTest.php
@@ -24,6 +24,21 @@ class ObjectTranslatorTest extends TestCase
 
         $this->assertSame('value3', $object->prop3());
     }
+
+    public function testCanTranslateProperties()
+    {
+        $object = new TranslatedObject(new ObjectForTranslationStub(), [
+            'prop1' => 'translated1',
+            'prop2' => 'translated2',
+            'prop3' => 'translated3',
+        ]);
+
+        $this->assertSame('translated1', $object->prop1);
+        $this->assertTrue(isset($object->prop1), 'Public property should be accessible');
+        $this->assertFalse(isset($object->prop2), 'Private property should not be accessible');
+        $this->assertSame('translated2', $object->prop2());
+        $this->assertSame('translated3', $object->getProp3());
+    }
 }
 
 class ObjectForTranslationStub
