diff --git a/templates/article/index.html.twig b/templates/article/index.html.twig
index 82b5eed..d14ec28 100644
--- a/templates/article/index.html.twig
+++ b/templates/article/index.html.twig
@@ -6,15 +6,14 @@
         <h1 class="text-4xl lg:text-7xl text-white mt-3 lg:mt-6 mb-12 text-center">{{ 'article.index.slogan'|trans }}</h1>
         <div>
             {% for article in articles %}
-            {% set article = article|translate_object %}
             <div class="bg-white my-2 rounded-2xl p-5">
                 <a href="{{ path('app_article_show', {slug: article.slug}) }}">
                     <div class="flex flex-col md:flex-row">
                         <img class="h-[206px] md:h-[412px] rounded-2xl" src="{{ asset('images/%s.png'|format(article.slug)) }}">
                         <div class="article-title d-inline-block pl-3 align-middle">
                             <div class="text-[#E53384] font-bold mt-8"> {{ article.publishedAt|date('F j, Y') }}</div>
-                            <div class="text-5xl font-medium mt-4">{{ article.title }}</div>
-                            <div class="line-clamp-3 my-5">{{ article.content|markdown_to_html }}</div>
+                            <div class="text-5xl font-medium mt-4">{{ article|translate_object.title }}</div>
+                            <div class="line-clamp-3 my-5">{{ article|translate_object.content|markdown_to_html }}</div>
                             <div class="flex items-center">
                                 <img class="rounded-full mr-1.5 h-10 w-10" src="{{ asset('images/alien-profile.png') }}">
                                 <div>{{ article.author }}</div>
