diff --git a/object-translation-bundle/src/ObjectTranslator.php b/object-translation-bundle/src/ObjectTranslator.php
index 0127616..1209e21 100644
--- a/object-translation-bundle/src/ObjectTranslator.php
+++ b/object-translation-bundle/src/ObjectTranslator.php
@@ -4,6 +4,8 @@ namespace SymfonyCasts\ObjectTranslationBundle;
 
 use Doctrine\Persistence\ManagerRegistry;
 use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+use Symfony\Contracts\Cache\TagAwareCacheInterface;
 use Symfony\Contracts\Translation\LocaleAwareInterface;
 use SymfonyCasts\ObjectTranslationBundle\Mapping\Translatable;
 use SymfonyCasts\ObjectTranslationBundle\Model\Translation;
@@ -67,7 +69,11 @@ final class ObjectTranslator
 
         return $this->cache->get(
             "object_translation.{$locale}.{$type}.{$id}",
-            function() use ($locale, $type, $id) {
+            function(ItemInterface $item) use ($locale, $type, $id) {
+                if ($this->cache instanceof TagAwareCacheInterface) {
+                    $item->tag(['object-translation', "object-translation-{$type}"]);
+                }
+
                 /** @var Translation[] $translations */
                 $translations = $this->doctrine->getRepository($this->translationClass)->findBy([
                     'locale' => $locale,
