diff --git a/object-translation-bundle/config/services.php b/object-translation-bundle/config/services.php
index 50a5a8f..c426385 100644
--- a/object-translation-bundle/config/services.php
+++ b/object-translation-bundle/config/services.php
@@ -3,6 +3,7 @@
 namespace Symfony\Component\DependencyInjection\Loader\Configurator;
 
 use SymfonyCasts\ObjectTranslationBundle\Command\ObjectTranslationExportCommand;
+use SymfonyCasts\ObjectTranslationBundle\Command\ObjectTranslationImportCommand;
 use SymfonyCasts\ObjectTranslationBundle\Command\ObjectTranslationWarmupCommand;
 use SymfonyCasts\ObjectTranslationBundle\ObjectTranslator;
 use SymfonyCasts\ObjectTranslationBundle\TranslatableMappingManager;
@@ -41,6 +42,12 @@ return static function (ContainerConfigurator $container): void {
             ])
             ->tag('console.command')
 
+        ->set('.symfonycasts.object_translator.import_command', ObjectTranslationImportCommand::class)
+            ->args([
+                service('.symfonycasts.object_translator.mapping_manager'),
+            ])
+            ->tag('console.command')
+
         ->set('.symfonycasts.object_translator.twig_extension', ObjectTranslatorExtension::class)
             ->tag('twig.extension')
     ;
diff --git a/object-translation-bundle/src/Command/ObjectTranslationImportCommand.php b/object-translation-bundle/src/Command/ObjectTranslationImportCommand.php
new file mode 100644
index 0000000..1099584
--- /dev/null
+++ b/object-translation-bundle/src/Command/ObjectTranslationImportCommand.php
@@ -0,0 +1,67 @@
+<?php
+
+namespace SymfonyCasts\ObjectTranslationBundle\Command;
+
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use SymfonyCasts\ObjectTranslationBundle\TranslatableMappingManager;
+
+/**
+ * @internal
+ */
+#[AsCommand(
+    name: 'object-translation:import',
+    description: 'Imports object translations from a CSV.',
+)]
+final class ObjectTranslationImportCommand extends Command
+{
+    public function __construct(
+        private TranslatableMappingManager $mappingManager,
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->addArgument('file', InputArgument::REQUIRED, 'The CSV file to import from.')
+            ->addArgument('locale', InputArgument::REQUIRED, 'The locale to import translations for.')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $file = $input->getArgument('file');
+        $locale = $input->getArgument('locale');
+
+        $io = new SymfonyStyle($input, $output);
+
+        $io->title('Importing Object Translations');
+
+        $fp = fopen($file, 'r');
+
+        $io->progressStart();
+
+        fgetcsv($fp);
+
+        while (($row = fgetcsv($fp)) !== false) {
+            [$type, $id, $field, $value] = $row;
+
+            $this->mappingManager->upsert($type, $id, $locale, $field, $value);
+
+            $io->progressAdvance();
+        }
+
+        fclose($fp);
+
+        $io->progressFinish();
+
+        $io->success(sprintf('Imported "%s"', $file));
+
+        return self::SUCCESS;
+    }
+}
diff --git a/object-translation-bundle/src/TranslatableMappingManager.php b/object-translation-bundle/src/TranslatableMappingManager.php
index babe320..4cc8229 100644
--- a/object-translation-bundle/src/TranslatableMappingManager.php
+++ b/object-translation-bundle/src/TranslatableMappingManager.php
@@ -102,4 +102,29 @@ final class TranslatableMappingManager
             yield $property->getName() => $property->getValue($object);
         }
     }
+
+    public function upsert(string $type, string $id, string $locale, string $field, string $value): void
+    {
+        $om = $this->doctrine->getManagerForClass($this->translationClass);
+
+        $translation = $om->getRepository($this->translationClass)->findOneBy([
+            'objectType' => $type,
+            'objectId' => $id,
+            'locale' => $locale,
+            'field' => $field,
+        ]);
+
+        if (!$translation) {
+            $translation = new ($this->translationClass)();
+            $translation->objectType = $type;
+            $translation->objectId = $id;
+            $translation->locale = $locale;
+            $translation->field = $field;
+        }
+
+        $translation->value = $value;
+
+        $om->persist($translation);
+        $om->flush();
+    }
 }
