diff --git a/templates/article/index.html.twig b/templates/article/index.html.twig
index 23f580a..b1fa5f1 100644
--- a/templates/article/index.html.twig
+++ b/templates/article/index.html.twig
@@ -7,7 +7,6 @@
         <div>
             {% for article in articles %}
             <div class="bg-white my-2 rounded-2xl p-5">
-                {{ 'hello_world'|trans }}
                 <a href="{{ path('app_article_show', {slug: article.slug}) }}">
                     <div class="flex flex-col md:flex-row">
                         <img class="h-[206px] md:h-[412px] rounded-2xl" src="{{ asset('images/%s.png'|format(article.slug)) }}">
diff --git a/translations/messages.en.yaml b/translations/messages.en.yaml
index 8653ee7..e69de29 100644
--- a/translations/messages.en.yaml
+++ b/translations/messages.en.yaml
@@ -1 +0,0 @@
-hello_world: "Hello, World"
diff --git a/translations/messages.fr.yaml b/translations/messages.fr.yaml
index 900c44d..e69de29 100644
--- a/translations/messages.fr.yaml
+++ b/translations/messages.fr.yaml
@@ -1 +0,0 @@
-hello_world: "Bonjour le monde!"
