diff --git a/templates/article/show.html.twig b/templates/article/show.html.twig
index 9a2dd20..fa49c0a 100644
--- a/templates/article/show.html.twig
+++ b/templates/article/show.html.twig
@@ -53,7 +53,7 @@
             <i class="text-[#0286C4] pl-1 fa-solid fa-share-nodes"></i>
         </div>
 
-        <h3 class="my-6 text-2xl font-medium"><i class="pr-3 fa fa-comment"></i>{{ 'article.show.comments'|trans({'%count%': 1}) }}</h3>
+        <h3 class="my-6 text-2xl font-medium"><i class="pr-3 fa fa-comment"></i>{{ 'article.show.comments'|trans({'%count%': comments|length}) }}</h3>
 
         <div class="flex items-center flex-shrink-0 mb-[40px]">
             <img class="rounded-full h-10 w-10" src="{{ asset('images/profile-img.png') }}">
