diff --git a/templates/base.html.twig b/templates/base.html.twig
index f23f245..058e5db 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -80,7 +80,7 @@
         </div>
         <footer class="py-8">
             <div class="text-white text-center">
-                {{ 'base.footer'|trans|raw }}
+                {{ 'base.footer'|trans({'%heart%': '<i class="fa fa-heart text-red-600"></i>', '%link%': '<a class="underline text-[#0286C4]" href="https://symfonycasts.com">SymfonyCasts</a>'})|raw }}
             </div>
         </footer>
     </body>
diff --git a/translations/messages.en.yaml b/translations/messages.en.yaml
index 0360305..4426f0e 100644
--- a/translations/messages.en.yaml
+++ b/translations/messages.en.yaml
@@ -1,6 +1,6 @@
 base:
     local_asteroids: "Local Asteroids"
-    footer: 'Made with <i class="fa fa-heart text-red-600"></i> by the guys and gals at <a class="underline text-[#0286C4]" href="https://symfonycasts.com">SymfonyCasts</a>'
+    footer: 'Made with %heart% by the guys and gals at %link%'
 
 article:
     show:
