diff --git a/templates/base.html.twig b/templates/base.html.twig
index 058e5db..11b8319 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -22,8 +22,8 @@
                     <img class="max-h-[45px]" src="{{ asset('images/spacebar-logo.png') }}" alt="Spacebar Logo">
                 </a>
                 <a href="#" class="hidden ml-4 mt-4 lg:inline-block lg:mt-0 hover:text-[#5BA8DC] mr-4">{{ 'base.local_asteroids'|trans }}</a>
-                <a href="#" class="hidden ml-4 mt-4 lg:inline-block lg:mt-0 hover:text-[#5BA8DC] mr-4">Weather</a>
-                <a href="#" class="hidden ml-4 mt-4 lg:inline-block lg:mt-0 hover:text-[#5BA8DC] mr-4">Journal</a>
+                <a href="#" class="hidden ml-4 mt-4 lg:inline-block lg:mt-0 hover:text-[#5BA8DC] mr-4">{{ 'base.weather'|trans }}</a>
+                <a href="#" class="hidden ml-4 mt-4 lg:inline-block lg:mt-0 hover:text-[#5BA8DC] mr-4">{{ 'base.journal'|trans }}</a>
             </div>
             <div class="flex items-center">
                 <div
