diff --git a/translations/messages.en.xliff b/translations/messages.fr.xliff
similarity index 95%
rename from translations/messages.en.xliff
rename to translations/messages.fr.xliff
index f879267..1246956 100644
--- a/translations/messages.en.xliff
+++ b/translations/messages.fr.xliff
@@ -7,7 +7,7 @@
     <body>
       <trans-unit id="SWPFW5B" resname="base.local_asteroids">
         <source>base.local_asteroids</source>
-        <target>Local Asteroids</target>
+        <target>(French) Local Asteroids</target>
       </trans-unit>
       <trans-unit id="mKAqmQl" resname="base.footer">
         <source>base.footer</source>
