diff --git a/README.md b/README.md
index f214eb0..b31143d 100644
--- a/README.md
+++ b/README.md
@@ -36,13 +36,10 @@ symfony composer install
 
 **Setup the Database**
 
-Create the database (SQLite by default, but if you want to go with
-a different DB server - configure the `DATABASE_URL` env var in the `.env`
-file first), then generate a migration, migrate, and load the fixtures:
+The app is all setup with an SQLite database and migrations. Reset
+the database and load the fixtures with:
 
 ```bash
-symfony console doctrine:database:create
-symfony console make:migration
 symfony console foundry:load-fixtures
 ```
 
diff --git a/migrations/Version20250728205442.php b/migrations/Version20250728205442.php
new file mode 100644
index 0000000..5cf2999
--- /dev/null
+++ b/migrations/Version20250728205442.php
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DoctrineMigrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20250728205442 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return '';
+    }
+
+    public function up(Schema $schema): void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->addSql('CREATE TABLE article (id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, category_id INTEGER NOT NULL, slug VARCHAR(255) NOT NULL, title VARCHAR(255) NOT NULL, published_at DATETIME NOT NULL --(DC2Type:datetime_immutable)
+        , author VARCHAR(255) NOT NULL, content CLOB NOT NULL, CONSTRAINT FK_23A0E6612469DE2 FOREIGN KEY (category_id) REFERENCES category (id) NOT DEFERRABLE INITIALLY IMMEDIATE)');
+        $this->addSql('CREATE INDEX IDX_23A0E6612469DE2 ON article (category_id)');
+        $this->addSql('CREATE TABLE category (id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, name VARCHAR(255) NOT NULL)');
+        $this->addSql('CREATE TABLE tag (id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, name VARCHAR(255) NOT NULL)');
+        $this->addSql('CREATE TABLE tag_article (tag_id INTEGER NOT NULL, article_id INTEGER NOT NULL, PRIMARY KEY(tag_id, article_id), CONSTRAINT FK_300B23CCBAD26311 FOREIGN KEY (tag_id) REFERENCES tag (id) ON DELETE CASCADE NOT DEFERRABLE INITIALLY IMMEDIATE, CONSTRAINT FK_300B23CC7294869C FOREIGN KEY (article_id) REFERENCES article (id) ON DELETE CASCADE NOT DEFERRABLE INITIALLY IMMEDIATE)');
+        $this->addSql('CREATE INDEX IDX_300B23CCBAD26311 ON tag_article (tag_id)');
+        $this->addSql('CREATE INDEX IDX_300B23CC7294869C ON tag_article (article_id)');
+    }
+
+    public function down(Schema $schema): void
+    {
+        // this down() migration is auto-generated, please modify it to your needs
+        $this->addSql('DROP TABLE article');
+        $this->addSql('DROP TABLE category');
+        $this->addSql('DROP TABLE tag');
+        $this->addSql('DROP TABLE tag_article');
+    }
+}
